src/test/java/io/jenkins/plugins/file_parameters/FileParameterWrapperTest.java

```diff
@@ -58,4 +62,153 @@
+        p.addProperty(new ParametersDefinitionProperty(new Base64FileParameterDefinition("FILE", null)));
+        String pipeline = "pipeline {\n" +
+            "  agent any\n" +
+            "  parameters {\n" +
+            "    base64File(name:'FILE')\n" +
+            "  }\n" +
+            "  stages {\n" +
+            "    stage('Example') {\n" +
+            "      steps {\n" +
+            "        withFileParameter(name:'FILE', allowNoFile: true) {\n" +
+            "          echo('foo') \n" +
+            "        }\n" +
+            "      }\n" +
+            "    }\n" +
+            "  }\n" +
+            "}";
+        p.setDefinition(new CpsFlowDefinition(pipeline, true));
+        WorkflowRun run = p.scheduleBuild2(0).get();
+        r.waitForCompletion(run);
+        r.assertBuildStatus(Result.SUCCESS, run);
+        r.assertLogContains("foo", run);
+        r.assertLogContains("Skip file parameter as there is no parameter with name: 'FILE'", run);
+    }
+
+    @Test public void base64DeclarativeParameterCreated() throws Exception {
+        r.createSlave("remote", null, null);
+        WorkflowJob p = r.createProject(WorkflowJob.class, "myjob");
+
+        String pipeline = "pipeline {\n" +
+            "  agent any\n" +
+            "  parameters {\n" +
+            "    base64File (name: 'FILE')\n" +
+            "  }\n" +
+            "  stages {\n" +
+            "    stage('Example') {\n" +
+            "      steps {\n" +
+            "        withFileParameter('FILE') {\n" +
+            "          echo(/loaded '${readFile(FILE).toUpperCase(Locale.ROOT)}' from $FILE/) \n" +
+            "        }\n" +
+            "      }\n" +
+            "    }\n" +
+            "  }\n" +
+            "}";
+
+        p.setDefinition(new CpsFlowDefinition(pipeline, true));
+        WorkflowRun run = p.scheduleBuild2(0).get();
+        r.waitForCompletion(run);
+        // definitely will fail but we just ensure parameter has been created
+        r.assertBuildStatus(Result.FAILURE, run);
+        ParametersDefinitionProperty pdp = p.getProperty(ParametersDefinitionProperty.class);
+        assertNotNull("parameters definition property is null", pdp);
+        ParameterDefinition pd = pdp.getParameterDefinition( "FILE");
+        assertNotNull("parameters definition is null", pd);
+        assertEquals("parameter not type Base64FileParameterDefinition", Base64FileParameterDefinition.class, pd.getClass());
+
+        assertThat(new CLICommandInvoker(r, "build").
+                       withStdin(new ByteArrayInputStream("uploaded content here".getBytes())).
+                       invokeWithArgs("-f", "-p", "FILE=", "myjob"),
+                   CLICommandInvoker.Matcher.succeeded());
+        WorkflowRun b = p.getBuildByNumber(2);
+        assertNotNull(b);
+        r.assertLogContains("loaded 'UPLOADED CONTENT HERE'", b);
+    }
+
+    @Test public void stashed() throws Exception {
```

Comment:
Fine but it is not using `FileParameterWrapper` so does not belong here. (Unless the two are made to be compatible and the test made to exercise that.)